src/Haskell/Ide/Engine/Monad.hs

```diff
@@ -46,6 +47,9 @@ data IdeState = IdeState
 
 runIdeM :: IdeState -> IdeM a -> IO a
 runIdeM initState f = do
+    case validatePlugins (idePlugins initState) of
```

Comment:
I think the plugin validation should probably happen in `MainHie`, before it starts spawning anything off.

So if there is a plugin error, it does not even start. And one day we will be asked to add a flag to skip the test, or run anyway even if there are errors. But I would say terminate on failure for now.